src/main/java/org/reaktivity/nukleus/tcp/internal/writer/stream/StreamFactory.java

```diff
@@ -186,10 +207,33 @@ private void doCleanup()
 
         private int handleWrite()
         {
-            // TODO: partial write completed
+            ByteBuffer writeBuffer = writeSlab.get(slot);
+            if (writeBuffer == null)
+            {
+                return 0;
```

Comment:
It used to be able to happen if processData results in a partial write, then processData is called again because more data comes in, and succeeds in writing out all the data (that remaining from the previous write plus the new data) before handleWrite gets calls by the selector.selectNow call. I have now fixed this by unsetting OP_WRITE in processData if this condition arises, so I will remove the if null condition (so it will do a hard fail with NullPointerException if the condition is encountered).